doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -433,6 +433,18 @@ implementation, the tactic will always choose the first possible
 solution to the set of constraints generated by a rewrite and will not
 try to find *all* the possible solutions to warn the user about them.
 
+Rewriting with ``Type`` valued relations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All definitions found in ``Classes.Relations``, ``Classes.Morphisms`` and
+``Classes.Equivalence``. are based on ``Prop``. Analogous definitions based on
+``Type`` can be found in ``Classes.CRelations``, ``Classes.CMorphisms`` and
+``Classes.CEquivalence`` respectively. The ``C`` stands for the "computational"
+versions. These definitions will have the same names as before.
+
+Importing these modules allows for generalized rewriting along relations
+``R : A -> A -> Type`` with support for universe polymorphism.
```

Comment:
```suggestion
Definitions in ``Classes.Relations``, ``Classes.Morphisms`` and
``Classes.Equivalence`` are based on ``Prop``. Analogous definitions 
with the same names based on
``Type`` are in ``Classes.CRelations``, ``Classes.CMorphisms`` and
``Classes.CEquivalence``. The ``C`` identifies the "computational"
versions.

Importing these modules allows for generalized rewriting along relations
``R : A -> A -> Type`` with support for universe polymorphism.
```

I don't understand the bit about "computational" versions--when is the
distinction important and why is it termed "computational"?

"along relations" -- is that the common wording?  How about the simpler
wording "using relations"?

Reply:
The "computational" aspect is because Type valued relations don't get erased during extraction, which is basically what Prop is there for. @mattam82 might be able to elaborate on the choice of name a bit further however (since he named it I believe).

>"along relations" -- is that the common wording? How about the simpler
wording "using relations"?

I think changing it to "with" might be best.

Reply:
"with" makes sense.  The phrasing should probably be "... with relations in the form \``R : A -> A -> Type`` ...".